plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -210,16 +210,22 @@ public synchronized void createDatabase(Database database)
     }
 
     @Override
-    public synchronized void dropDatabase(String databaseName, boolean deleteData)
+    public synchronized void dropDatabase(String databaseName, boolean deleteData, boolean cascade)
     {
         requireNonNull(databaseName, "databaseName is null");
 
         // Database names are stored lowercase. Accept non-lowercase name for compatibility with HMS (and Glue)
         databaseName = databaseName.toLowerCase(ENGLISH);
 
         getRequiredDatabase(databaseName);
-        if (!getAllTables(databaseName).isEmpty()) {
-            throw new TrinoException(HIVE_METASTORE_ERROR, "Database " + databaseName + " is not empty");
+        List<String> tables = getAllTables(databaseName);
+        if (!tables.isEmpty()) {
+            if (!cascade) {
+                throw new TrinoException(HIVE_METASTORE_ERROR, "Database " + databaseName + " is not empty");
+            }
+            for (String tableName : tables) {
```

Comment:
Just making sure here - in the situation in which the `FileHiveMetastore` (used for testing AFAIK) does a `DROP SCHEMA CASCADE` whille concurrently adding a new table there may be a corruption of the metastore. However since `FileHiveMetastore` is used for tests only this should be acceptable. Right?

Reply:
Right.